### DISCUSSION TITLE
Set tags for multiple entries

### DISCUSSION THREAD

**Discussion by apraga** [papis/papis#727] [Q&A] [answered]
Hi,

What is the best strategy to add or edit tags for multiple entries ?
So far, I've found 2 ways of doing it 
1. With `papis edit`, mark entries with `Ctrl-T` and edit the yaml for each entry (cumbersome).
2. If entries can be queried (e.g there share part of the title), `papis update KEYWORD -s tag MYTAG` works. One caveat is that it does not seem to append the tags but instead replace them.

I feel this workflow is missing from the documentation (or is not easily discoverable). Anyway, I've switched to papis full-time so am looking for low-friction workflows :)

Thanks,

**Comment by alexfikl** [comment, marked as answer]
Hi! Thank you for your interest in `papis` :grin: 

Unfortunately at the moment there is no convenient workflow for working with tags. There's some ongoing work in #681 and #648 that should add some pretty nifty and flexible support for exactly what you're asking for, but it's not quite done yet.

In the meantime, papis is very pluggable, so you can just write a `papis tag` command for yourself to best suit your workflow. There's an example of a standalone script that plays around with tags here
https://github.com/papis/papis/blob/04f6b62bca4c84b4ad4d3eb97fc0d7d4742a3ba5/examples/scripts/papis-tags
and you can also take a look at the existing commands and write a separate plugin package yourself. For an example of how to set up the `setup.py` file for that, you can take a look at `papis-zotero` that adds a `papis zotero` subcommand
https://github.com/papis/papis-zotero

There's some (pretty bare) documentation about plugins here
https://papis.readthedocs.io/en/latest/plugins.html

Hope that helps a bit!

**Reply by alexfikl** [reply]
Ah, for the standalone scripts, this is documented over here
https://papis.readthedocs.io/en/latest/scripting.html

**Reply by apraga** [reply]
Thanks for linking to these issues that adress point number 2. Once implemented, it would be great if the update could happen after marking entries too :)

**Reply by apraga** [reply]
#648 has been merged, so marking it as answered. Good work !
